File: 1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        return dfs(original,cloned,target);
    }
    
    private TreeNode dfs(TreeNode orginal, TreeNode copy, TreeNode target){
        if(orginal == null || copy == null) return null;
        if(target == orginal && orginal.val == copy.val) return copy;
        
        TreeNode left = dfs(orginal.left,copy.left, target);
        TreeNode right = dfs(orginal.right, copy.right, target);
        
        return (left == null) ? right : left; 
        
    }
}

File: 1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:
        
        
        def dfs(root1, root2, target):
            if(not root1):
                return
            if(root1 == target):
                return root2
            
            left = dfs(root1.left,root2.left,target)
            if(left):
                return left
            return dfs(root1.right,root2.right,target)
        
                
        return dfs(original,cloned,target)